contracts/factory/K1ValidatorFactory.sol

```diff
@@ -80,14 +82,7 @@ contract K1ValidatorFactory is Stakeable {
         uint8 threshold
     ) external payable returns (address payable) {
         // Compute the actual salt for deterministic deployment
-        bytes32 actualSalt;
-        assembly {
-            let ptr := mload(0x40)
-            let calldataLength := sub(calldatasize(), 0x04)
-            mstore(0x40, add(ptr, calldataLength))
-            calldatacopy(ptr, 0x04, calldataLength)
-            actualSalt := keccak256(ptr, calldataLength)
-        }
```

Comment:
not quite sure why these two are not the same?

I believe this change was done only for gas optimisation from gaslite suggestion.

Reply:
you're effectively calculating `keccak256(abi.encode(...params))` while his code is `keccak256(abi.encodePacked(...params)`. Generally `encodePacking` is cheaper but I can't say for sure in this case if re-encoding will be cheaper than copying and hashing pre-encoded data.

Reply:
check how arrays are abi encoded, there's an offset to the data (that can be chosen by the caller as the issue says) and you'd be encoding this offset too along with everything else.